server/workflowarchive/archived_workflow_server.go

```diff
@@ -291,12 +292,25 @@ func (w *archivedWorkflowServer) RetryArchivedWorkflow(ctx context.Context, req
 			return nil, sutils.ToStatusError(err, codes.Internal)
 		}
 
+		errCh := make(chan error, len(podsToDelete))
+		var wg sync.WaitGroup
+		wg.Add(len(podsToDelete))
 		for _, podName := range podsToDelete {
```

Comment:
It feels like this and the partner could be extracted out to a common helper function.

Reply:
Yes, this is possible. I can extract a function.